app/controllers/report_controller.rb

```diff
@@ -1,81 +1,42 @@
 class ReportController < ApplicationController
   protect_from_forgery with: :null_session
+  before_action :authorize_anonymous, :only=>[:add_record]
 
   def show
     @user = User.find_by(username: params[:name])
     if @user
-      @reports = Report.where("user_id = '#{@user&.id}'")
+      @reports = @user.reports
     else
       @reports = Report.where("user_id IS ?", nil)
     end
   end
 
   def add_record
-    auth = authorize_request
-    if auth[:errors]
-      render json: "Authorization token invalid: #{auth[:errors]}"
-      return
-    end
-
     data = JSON.parse(request.body.read)
-    user = auth[:user]
-    device = Device.find_by(device_id: data['user_id'])
+    device = Device.find_by(uuid: data['user_id'])
     if !device
-      device = Device.new(device_id: data['user_id'],
-                          user_id: user&.id
+      device = Device.new(uuid: data['user_id'],
```

Comment:
Binding user id and device id doesn't quite make sense to me. Assuming several users use AWS EC2 t2.medium device, are these devices going to be recorded as one single entry or different entries in the database?

Reply:
So there's two things happening on this line which may not be very clear.

* The carbon client sends us an ID field as part of its payload. This is a unique string identifying the device it was sent from and is used as the primary key in the table of devices. It isn't simply the name of the instance (e.g. `t2.medium`), it's a long sequence of characters that is unique to each device.
* In the original version of carbon client it was the UUID of the device, which I thought made sense to call `user_id`. It turns out that giving it that name was a bad idea, but it's still called that in the current version, so this line retrieves `data['user_id']` which is completely distinct from any IDs found in the table of users. Once Stu's work on the carbon client gets merged, I believe this field will be renamed to something like `device_id` which is far more appropriate for what it is.

Reply:
For the first point, if there are several t2.medium instances, their `user_id` or `device_id` will be the same isn't it?

Reply:
If those instances are all being hosted on the same device then yes. However I believe the latest PR for carbon client that was just raised addresses this and ensures that the device IDs are ALWAYS unique.